extensions/notebook/src/jupyter/jupyterServerInstallation.ts

```diff
@@ -714,9 +711,28 @@ export class JupyterServerInstallation implements IJupyterServerInstallation {
 		return this._pythonExecutable;
 	}
 
-	public getCondaExePath(): string {
-		return path.join(this._pythonInstallationPath,
-			process.platform === constants.winPlatform ? 'Scripts\\conda.exe' : 'bin/conda');
+	public get condaExecutable(): string | undefined {
+		return this._condaExecutable;
+	}
+
+	public static async getCondaExePath(pythonInstallationPath: string): Promise<string> {
+		let exeName = process.platform === constants.winPlatform ? 'Scripts\\conda.exe' : 'bin/conda';
+		let condaPath = path.join(pythonInstallationPath, exeName);
+		let condaExists = await fs.pathExists(condaPath);
+		// If conda was not found, then check if we're using a virtual environment
```

Comment:
Would suggest adding a few logging statements here for the searching to help with debugging further issues with conda environments.

Reply:
These path issues tend to be pretty obvious. I also don't want to pollute the log with "conda not found" messages for people using regular python.